src/modules/Transcript/index.tsx

```diff
@@ -166,19 +169,37 @@ export const TranscriptUI = (props: TranscriptUIProps): ReactElement => {
-                                                            >
-                                                                {text}
-                                                            </mark>
-                                                        ) : (
-                                                            <span key={id}>{text}</span>
+                                                    {sentences.map(({ chunks }) =>
+                                                        chunks.map(
+                                                            ({ highlight, id: sentenceId, text, textSentiment }) =>
+                                                                highlight ? (
+                                                                    <mark
+                                                                        ref={
+                                                                            sentenceId === currentMatch
+                                                                                ? currentMatchRef
+                                                                                : undefined
+                                                                        }
+                                                                        className={classNames({
+                                                                            'bg-yellow-300':
+                                                                                sentenceId === currentMatch,
+                                                                        })}
+                                                                        key={sentenceId}
+                                                                    >
+                                                                        {text}
+                                                                    </mark>
+                                                                ) : (
+                                                                    <span
+                                                                        key={sentenceId}
+                                                                        className={classNames({
+                                                                            'text-green-600':
+                                                                                textSentiment === 'positive',
+                                                                            'text-red-600':
+                                                                                textSentiment === 'negative',
+                                                                        })}
+                                                                    >
+                                                                        {/(\.|\?|!)$/.exec(text) ? `${text} ` : text}
```

Comment:
wow this is some shit! in my code it was like text.match(/...) shit got completely re-written like I'm a baby or some shit

Reply:
I don't get it lol. What is this doing?